### PR TITLE
chore: code review

### PR DESCRIPTION
**Pull request by thepassle**

### CODE REVIEW

File: lib/rules/scoped-elements.js [hunk 1]
```diff
@@ -46,7 +46,6 @@ module.exports = {
        * This will check lit-html tagged template literals
        */
 
-      //maybe better with map filter reduce?
```
Comment (thepassle) [inline lib/rules/scoped-elements.js:49]: for/of loops are fine 🙂

File: tests/lib/rules/scoped-elements.js [hunk 2]
```diff
@@ -132,14 +132,38 @@ class MyElement extends ScopedElementsMixin(LitElement) {
...
+}
+    `,
+        errors: [{ message: 'Element not declared in static scopedElements' }]
+      },
+      {
+        /** ❌ Not declared */
+        code: `
+class MyElement extends ScopedElementsMixin(LitElement) {
+connectedCallback() {
+  super.connectedCallback();
```
Comment (thepassle) [inline tests/lib/rules/scoped-elements.js:149]: Currently in your code, if there is ANY kind of `super.whatever()` call present, it will ignore linting scoped elements. So also if you use `super.connectedCallback()`, it'll skip linting scoped elements. Added a test to catch this 🙂

File: tests/lib/rules/scoped-elements.js [hunk 3]
```diff
@@ -132,14 +132,38 @@ class MyElement extends ScopedElementsMixin(LitElement) {
...
+render() { return html\`<my-el></my-el>\`}
+}
+    `,
+        errors: [{ message: 'Element not declared in static scopedElements' }]
+      },      
         {
-            /** ❌ Not declared */
             code: `
 class MyElement extends ScopedElementsMixin(LitElement) {
+  get scopedElements() {
```
Comment (thepassle) [inline tests/lib/rules/scoped-elements.js:159]: You're not checking for `static get scopedElements`, you're checking for `get scopedElements`, which in theory should be super edge-casey (I dont expect anybody to use `get scopedElements`, but someone might). Just goes to show how specific you have to be when writing linting plugins. Added a failing test to catch it

**Comment by thepassle** [comment]
Another thing (that I didnt add a failing test for):
You're currently skipping linting for scopedElements when the following code is present in the file:
```js
this.defineScopedElement('my-el', m.MyEl)
```
This kind of scopedElement definition should be fine (and not skip linting for scopedElements), because the first argument of `defineScopedElement` is just a string, so we can totally check for this. So make sure to not set `hasDynamicValuesInDefineScopedElementsCall` to true for this scenario.

The scenario we want to skip is the following:
```js
this.defineScopedElement(someVariable, m.MyEl);
```
Since `someVariable` is dynamic (can be imported from anywhere, changed at runtime, etc), this is hard/not possible to statically analyze. We only want to set `hasDynamicValuesInDefineScopedElementsCall` to `true` in this scenario.
